Source/WebKit/Shared/WebCoreArgumentCoders.cpp

```diff
@@ -364,7 +256,7 @@ void ArgumentCoder<DOMCacheEngine::Record>::encode(Encoder& encoder, const DOMCa
 
     WTF::switchOn(record.responseBody, [&](const Ref<SharedBuffer>& buffer) {
```

Comment:
so the consistent manner:
"all encoders should do is just encoder << .... << .. <<. ..;"

so here you could do just
`encoder << record.responseBody;
`

Reply:
I'll take those in the follow-up patch

Reply:
I'm puzzled with those on how to perform the decoding side.
    auto responseBody = decoder.decode<WebCore::DOMCacheEngine::ResponseBody>();

gives:
```
In file included from /Users/jyavenard/Work/webkit/OpenSource/WebKitBuild/Debug/DerivedSources/WebKit/unified-sources/UnifiedSource41.cpp:7:
/Users/jyavenard/Work/webkit/OpenSource/Source/WebKit/Shared/WebCoreArgumentCoders.cpp:298:33: note: in instantiation of function template specialization 'IPC::Decoder::decode<std::variant<std::nullptr_t, WTF::Ref<WebCore::FormData>, WTF::Ref<WebCore::SharedBuffer>>>' requested here
    auto responseBody = decoder.decode<WebCore::DOMCacheEngine::ResponseBody>();
                                ^
In file included from /Users/jyavenard/Work/webkit/OpenSource/WebKitBuild/Debug/DerivedSources/WebKit/unified-sources/UnifiedSource41.cpp:1:
In file included from /Users/jyavenard/Work/webkit/OpenSource/Source/WebKit/Shared/WebCompiledContentRuleList.cpp:27:
In file included from /Users/jyavenard/Work/webkit/OpenSource/Source/WebKit/Shared/WebCompiledContentRuleList.h:30:
In file included from /Users/jyavenard/Work/webkit/OpenSource/Source/WebKit/Shared/WebCompiledContentRuleListData.h:30:
In file included from /Users/jyavenard/Work/webkit/OpenSource/Source/WebKit/Platform/SharedMemory.h:31:
In file included from /Users/jyavenard/Work/webkit/OpenSource/WebKitBuild/Debug/usr/local/include/wtf/ThreadSafeRefCounted.h:30:
In file included from /Users/jyavenard/Work/webkit/OpenSource/WebKitBuild/Debug/usr/local/include/wtf/MainThread.h:35:
In file included from /Users/jyavenard/Work/webkit/OpenSource/WebKitBuild/Debug/usr/local/include/wtf/ThreadingPrimitives.h:37:
In file included from /Users/jyavenard/Work/webkit/OpenSource/WebKitBuild/Debug/usr/local/include/wtf/WallTime.h:29:
In file included from /Users/jyavenard/Work/webkit/OpenSource/WebKitBuild/Debug/usr/local/include/wtf/GenericTimeMixin.h:28:
In file included from /Users/jyavenard/Work/webkit/OpenSource/WebKitBuild/Debug/usr/local/include/wtf/Seconds.h:28:
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.0.Internal.sdk/usr/include/c++/v1/optional:694:41: note: candidate constructor not viable: no known conversion from 'std::optional<FormData>' to 'const std::optional<WTF::Ref<WebCore::FormData>> &' for 1st argument
    _LIBCPP_INLINE_VISIBILITY constexpr optional(const optional&) = default;
                                        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.0.Internal.sdk/usr/include/c++/v1/optional:695:41: note: candidate constructor not viable: no known conversion from 'std::optional<FormData>' to 'std::optional<WTF::Ref<WebCore::FormData>> &&' for 1st argument
    _LIBCPP_INLINE_VISIBILITY constexpr optional(optional&&) = default;
                                        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.0.Internal.sdk/usr/include/c++/v1/optional:696:41: note: candidate constructor not viable: no known conversion from 'std::optional<FormData>' to 'std::nullopt_t' for 1st argument
    _LIBCPP_INLINE_VISIBILITY constexpr optional(nullopt_t) noexcept {}
                                        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.0.Internal.sdk/usr/include/c++/v1/optional:720:15: note: candidate template ignored: substitution failure [with _Up = std::optional<WebCore::FormData>]: no member named '_EnableIfImpl' in 'std::_MetaBase<false>'
    constexpr optional(_Up&& __v)
              ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.0.Internal.sdk/usr/include/c++/v1/optional:735:35: note: candidate template ignored: substitution failure [with _Up = WebCore::FormData]: no member named '_EnableIfImpl' in 'std::_MetaBase<false>'
    _LIBCPP_CONSTEXPR_AFTER_CXX17 optional(const optional<_Up>& __v)
                                  ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.0.Internal.sdk/usr/include/c++/v1/optional:753:35: note: candidate template ignored: substitution failure [with _Up = WebCore::FormData]: no member named '_EnableIfImpl' in 'std::_MetaBase<false>'
    _LIBCPP_CONSTEXPR_AFTER_CXX17 optional(optional<_Up>&& __v)
                                  ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.0.Internal.sdk/usr/include/c++/v1/optional:706:24: note: explicit constructor is not a candidate
    constexpr explicit optional(_InPlaceT, _Args&&... __args)
                       ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.0.Internal.sdk/usr/include/c++/v1/optional:727:24: note: explicit constructor is not a candidate
    constexpr explicit optional(_Up&& __v)
                       ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.0.Internal.sdk/usr/include/c++/v1/optional:743:44: note: explicit constructor is not a candidate
    _LIBCPP_CONSTEXPR_AFTER_CXX17 explicit optional(const optional<_Up>& __v)
                                           ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.0.Internal.sdk/usr/include/c++/v1/optional:761:44: note: explicit constructor is not a candidate
    _LIBCPP_CONSTEXPR_AFTER_CXX17 explicit optional(optional<_Up>&& __v)
                                           ^
In file included from /Users/jyavenard/Work/webkit/OpenSource/WebKitBuild/Debug/DerivedSources/WebKit/unified-sources/UnifiedSource41.cpp:2:
In file included from /Users/jyavenard/Work/webkit/OpenSource/Source/WebKit/Shared/WebCompiledContentRuleListData.cpp:31:
In file included from /Users/jyavenard/Work/webkit/OpenSource/Source/WebKit/Platform/IPC/ArgumentCoders.h:28:
/Users/jyavenard/Work/webkit/OpenSource/Source/WebKit/Platform/IPC/ArgumentCoder.h:63:20: error: no viable conversion from returned value of type 'RefPtr<WebCore::FormData>' to function return type 'std::optional<FormData>'
            return T::decode(decoder);
                   ^~~~~~~~~~~~~~~~~~
```